spec/catalog.yaml

```diff
@@ -396,6 +440,126 @@ components:
+        - id
+        - logicalType
+      properties:
+        type:
+          type: string
+          enum:
+            - PARENT
+            - REPEATED
+            - LEAF
+        name:
+          type: string
+        id:
+          type: int
+        parentId:
+          type: int
+        logicalType:
+          type: string
+          description: >
+            Logical types, currently support parameterized Arrow Type.
+            
+            PARENT types will always have logical type "struct".
+            
+            Logical type "map" is represented as a struct with a single child field "entries".
+            "entries" is a struct with fields "key" and "value".
+
+            REPEATED types may have logical types:
+            * "list"
+            * "large_list"
+            * "list.struct"
+            * "large_list.struct"
+            The final two are used if the list values are structs, and therefore the
+            field is both implicitly REPEATED and PARENT.
+          
+            LEAF types may have logical types:
+            * "null"
+            * "bool"
+            * "int8" / "uint8"
+            * "int16" / "uint16"
+            * "int32" / "uint32"
+            * "int64" / "uint64"
+            * "halffloat" / "float" / "double"
+            * "string" / "large_string"
+            * "binary" / "large_binary"
+            * "date32:day"
+            * "date64:ms"
+            * "decimal:128:{precision}:{scale}" / "decimal:256:{precision}:{scale}"
+            * "time:{unit}" / "timestamp:{unit}" / "duration:{unit}", where unit is "s", "ms", "us", "ns"
+            * "dict:{value_type}:{index_type}:false"
+        nullable:
+          type: boolean
+        metadata:
+          type: object
+          description: optional field metadata (e.g. extension type name/parameters)
+          additionalProperties:
+            type: string
+
+    WriterVersion:
+      type: object
+      properties:
+        library:
+          type: string
+          description: The name of the library that created this file.
+        version:
+          type: string
+          description: >
+            The version of the library that created this file. 
+            Because we cannot assume that the library is semantically versioned, this is a string. 
+            However, if it is semantically versioned, it should be a valid semver string without any 'v' prefix. 
+            For example: `2.0.0`, `2.0.0-rc.1`.
+
+    CreateTableRequest:
+      type: object
+      required:
+        - name
+        - schema
+      properties:
+        name:
+          type: string
+        mode:
+          type: string
+          enum:
+            - CREATE
+            - EXIST_OK
+            - OVERWRITE
+          default: CREATE
+        location:
+          type: string
+        schema:
+          $ref: '#/components/schemas/Schema'
+        writerVersion:
+          $ref: '#/components/schemas/WriterVersion'
+        config:
```

Comment:
We might not be able to pass anything here yet with the current shape of the `Dataset` API. But I remember we still had discussion about the storage options last time. This is different from that as these will be persisted in the Lance table metadata. 

Do you still think it is necessary to pass storage options in as a part of the request? @yanghua

Reply:
> My understanding is that the server should decide the storage options used based on the location.

Our commercial version also uses mode.

My original thought is that maybe we can open an entry point for users, they can pass `ak/sk` or not, that depends themself.

OK, now let's ignore it now.

Reply:
passing sk is definitely a no go I think, because the server can easily just keep the sk to use outside the duration of the request and it is a huge security red flag. We probably need to solve that when implementing AuthN for the client and server.

Reply:
> passing sk is definitely a no go I think, because the server can easily just keep the sk to use outside the duration of the request and it is a huge security red flag

In our cases, that depends on how we define the `client`. A client may always be on a cloud vendor's intranet.
Or, in some cases, the customer may give us these informations to handle emergency situations.

Keeping it doesn't necessarily mean using it or encouraging it.

Anyway, let's ignore it. Currently, close this entry point to pass `storage options`.